sstables/sstable_set.cc

```diff
@@ -505,6 +565,19 @@ void time_series_sstable_set::insert(shared_sstable sst) {
   }
 }
 
+void time_series_sstable_set::insert(const std::vector<shared_sstable>& ssts) {
+    try {
+        for (const auto& sst : ssts) {
+            insert(sst);
+        }
+    } catch (...) {
+        for (const auto& sst : ssts) {
+            erase(sst);
+        }
+        throw;
+    }
+}
+
```

Comment:
because each sst can go into a different time window, so the simple loop here at the top of the hierarchy is trivial.
We can sort the sstables by time window and then insert them in batches by using ranges from the sorted vector, but I"m not sure it's worth it.
@raphaelsc WDYT?

Reply:
I meant, why can't a similar loop be used for `partitioned_sstable_set`?

Reply:
Well, it can. It'd just be slightly less efficient to insert the sstables one at a time.

Reply:
You know what, since this isn't this gist of this PR, I'll keep it simple and less risky and can do the vector sstable insert optimization as a followup patch.

Reply:
in v2 this function is the default implementation and it's used both for `time_series_sstable_set` and `partitioned_sstable_set`.